### PR TITLE
Feat/new password

### PR DESCRIPTION
**Pull request by zweroboy1**
## Describe your changes
- Implemented an authentication guard (JwtAuthGuard) to restrict access to routes for unauthenticated users.
- Created a new route for users to reset their forgotten passwords. This route is secured and can be accessed after successful verification of a one-time password (OTP).

## Issue ticket code (and/or) and link
- [Link to JIRA ticket](https://2024internship.atlassian.net/browse/SCRUM-48)
  
### **General**
- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [ ] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Backend
- [x] Swagger documentation updated
- [x] Database requests are optimized and not redundant
- [ ] Unit tests written
- [x] use ConfigService instead of process.env
- [ ] use transactions if there is a call chain that mutates data in different tables
- [ ] use @index decorator for frequently requested data

### CODE REVIEW

File: src/modules/auth/auth.service.ts [hunk 1]
```diff
@@ -211,9 +220,17 @@ export class AuthService {
...
+
     const publicUser = this.usersService.buildPublicUser(user);
-    const userWithTokens = await this.generateUserWithTokens(publicUser);
+    const verifiedUser = this.makeUserVerified(publicUser);
+    const userWithTokens = await this.generateUserWithTokens(verifiedUser);
 
     return userWithTokens;
   }
+
+  async changePassword(id: string, dto: PasswordDto): Promise<void> {
```
Comment (lyudad) [inline src/modules/auth/auth.service.ts]: its service why you pass full dto if only password used?

Comment (lyudad) [reply src/modules/auth/auth.service.ts]: and name dto does not mean noting, naming should be self explained

Comment (zweroboy1) [reply src/modules/auth/auth.service.ts]: For consistency across the project, I always passed the DTO from the controller to the service and destructure the DTO into variables within the service. The current PasswordDto contains only one field - {"password": "..."}. It is used for validating incoming data using class-validator through UsePipes. This approach is also consistent throughout our project, ensuring that any incoming JSON that fails validation shows an error to the user and does not proceed to the controller or to the service.

You’re right, now I only pass individual variables to the service, unless it’s a DTO with many fields. I also changed the names to be more meaningful.

**Comment by lyudad** [comment]
its very not good practice use DTO as type for service, as your input or

**Pull request closed without merging**